backend/repository/gsheet_repository.py

```diff
@@ -0,0 +1,86 @@
+import gspread
+import pandas as pd
+import numpy as np
+import os
+import logging
+
+from backend.repository import Repository
+
+log = logging.getLogger(__name__)
+
+
+class GSheetRepository(Repository):
```

Comment:
good idea to have a class to deal with (from what I understood) all update interactions with the google sheet

Reply:
good idea to preempt future data stores using the abstract base class as a contract

Reply:
Yeah this is an abstraction layer, to enact among other things [Separation of concerns](https://en.wikipedia.org/wiki/Separation_of_concerns). If all interactions with a certain technology or 3th party are all handled in a single location then it is very easy to swap out. And you do that by just exposing some basic functions as defined in the base class Repository in the above example...
if we create an SQLRepository... as long as it implements all the basic methods of Repository it can replace GSheetRepository with 1 line of code changes in the task.